aict-MERN/backend-of-aict-site-w-react/app.js

```diff
@@ -0,0 +1,24 @@
+import express from 'express'
+const app = express()
+import router from './routes/posts.js'
+import cors from 'cors';
+
+const corsOptions = {
+  origin: (origin, callback) => {
+    callback(null, true);
+  },
+  methods: ["GET", "POST", "PUT", "PATCH", "DELETE"],
+  allowedHeaders: ["Access-Control-Allow-Origin", "Origin", "X-Requested-With", "Content-Type", "Accept", "Authorization"],
+  credentials: true
+};
+
+app.options('*', cors(corsOptions));
+app.use(cors(corsOptions));
+
+app.use(express.json())
+app.use(express.urlencoded({ extended: true }))
+
+app.use('/', router)
+
+const port = process.env.PORT || 3001
+app.listen(port, console.log(`App listening on port ${port}`))
```

Comment:
nevermind. that console.log here isnt an anonymous function xD you forgot to add the `() =>`

Reply:
This is odd, that message still comes up, should it have the arrow function on there?
_

Reply:
so whats happening here is that the return type of a function that returns nothing is `undefined`, which is what `console.log` returns, and its also a valid value to pass to `app.listen`. so the `console.log` gets executed, because its called like any other function, then it returns `undefined` which is passed to `listen`, which then proceeds to ignore it completely.

Reply:
Ahh I see, so it doesn't really make a difference but it's better to throw in the arrow function?